### PR TITLE
docs: added code review guide

### PR DESCRIPTION
**Pull request by bhavsarpratik**

### CODE REVIEW

File: .github/CODE_REVIEW_GUIDELINES.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+## Jina specific  
+
+- Use this to create random ports for servers
+
+    `from jina.helper import random_port`
+
+- You can keep function specific imports inside functions
+- Jina ENV variables should start with **JINA_**
+- For type hints which require import, you can do like this
```
Comment (alexcg1) [inline .github/CODE_REVIEW_GUIDELINES.md:45]: do **it** like this

File: .github/CODE_REVIEW_GUIDELINES.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+- Add new unit & integration test if you add a new feature
+- Use pytest tmpdir fixture for temporary directories
+- Check test locally before pushing
+- If a `Flow()` needs to be built in the test and is not a test in the Flow module -> It belongs to integration test
+
+## Core  
+
+**Single Responsibility Principle (SRS)**  
+
+Do not place more than one responsibility into a single class or function, refactor into separate classes and functions.
```
Comment (alexcg1) [inline .github/CODE_REVIEW_GUIDELINES.md:67]: Do not place more than one responsibility into a single class or function, Instead, refactor into separate classes and functions.

File: .github/CODE_REVIEW_GUIDELINES.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+- Approve PR only if you are sure about the scope
+- Be respectful and reply asap
+- Do not merge PR without approval
+- Avoid spending too much time on trivial changes
+- Avoid premature optimisation
+- For community PR, check CLA by typing `recheckcla`
+- Ask for jina env if needed by `jina -vf`
+- Suggest changes and check back ASAP. Get PR merged soon
+- For new features, suggest adding appropriate documentation
+- Don&#39;t forget to praise when PR is ready with something like LGTM
```
Comment (cristianmtr) [inline .github/CODE_REVIEW_GUIDELINES.md:23]: what's this weird character? some encoding problem?

**Review by alexcg1 — CHANGES_REQUESTED** [review]
Some tiny feedback to improve legibility

File: .github/CODE_REVIEW_GUIDELINES.md (hunk 3 above)
Comment (bhavsarpratik) [reply .github/CODE_REVIEW_GUIDELINES.md:23]: Nice catch; I converted the doc to markdown through an online converter which does this for `'`

File: .github/CODE_REVIEW_GUIDELINES.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    from PIL import Image
+    ...
+```
+
+## Adding tests  
+
+- Add new unit & integration test if you add a new feature
+- Use pytest tmpdir fixture for temporary directories
+- Check test locally before pushing
+- If a `Flow()` needs to be built in the test and is not a test in the Flow module -> It belongs to integration test
```
Comment (cristianmtr) [inline .github/CODE_REVIEW_GUIDELINES.md:61]:
```suggestion
- If a `Flow()` needs to be built in the test and it's not a test in the Flow module, the test belongs under `/tests/integration`.
```

File: .github/CODE_REVIEW_GUIDELINES.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+While adding new functionality, existing code should not be modified. New functionality should be written in new classes and functions.
+
+**Liskov substitutability principle**  
+
+The child class should not change the behavior (meaning) of the parent class. The child class can be used as a substitute for a base class.
+
+**Interface segregation**  
+
+Do not create lengthy interfaces, instead split them into smaller interfaces based on the functionality. The interface should not contain any dependencies (parameters), which are not required for the expected functionality.
```
Comment (cristianmtr) [inline .github/CODE_REVIEW_GUIDELINES.md:79]:
```suggestion
Do not create lengthy interfaces. Instead split them into smaller interfaces based on the functionality. The interface should not contain any dependencies (parameters), which are not required for the expected functionality.
```

File: .github/CODE_REVIEW_GUIDELINES.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+**Dependency Injection**  
+
+Do not hardcode the dependencies, instead inject them.
```
Comment (cristianmtr) [inline .github/CODE_REVIEW_GUIDELINES.md:83]:
```suggestion
Do not hardcode the dependencies; instead inject them.
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -66,11 +66,12 @@ Right now we're working on a list of things we want help with and easy-to-fix bu
...
-5. Make a pull request. 
-6. Submit your pull request and wait for all checks to pass.
-7. Request reviews from one of [the code owners](.github/CODEOWNERS).
-8. Get a LGTM 👍 and PR gets merged.
+4. Check if your code changes follow the [code review guidelines](.github/CODE_REVIEW_GUIDELINES.md).
+5. Commit the changes with the [correct commit style](#writing-your-commit-message).
+6. Make a pull request. 
+7. Submit your pull request and wait for all checks to pass.
+8. Request reviews from one of [the code owners](.github/CODEOWNERS).
+9. Get a LGTM 👍 and PR gets merged.
```
Comment (cristianmtr) [inline CONTRIBUTING.md:74]: You can just write 

```
1. First item
1. Second item
```

and you get automatic listing.

1. First item
1. Second item

So we avoid having changes to the numbering of the items after the one you add.

**Review by cristianmtr — CHANGES_REQUESTED** [review]
Minor things

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -64,14 +64,14 @@ Right now we're working on a list of things we want help with and easy-to-fix bu
 
 0. Associate your local git config with your github account. If this is your first time using git you can follow [the steps](#associate-with-github-account).
```
Comment (cristianmtr) [inline CONTRIBUTING.md:65]:
```suggestion
1. Associate your local git config with your github account. If this is your first time using git you can follow [the steps](#associate-with-github-account).
```

Comment (bhavsarpratik) [reply CONTRIBUTING.md:65]: It was originally kept 0 and so I let it be to give special importance. The rest of the numbering is working fine.

**Review by bwanglzu — APPROVED** [review]
great work, only this line `Do not merge PR without approval`. we can't do it without approval because the PR is blocked lol

File: .github/CODE_REVIEW_GUIDELINES.md [hunk 9]
```diff
@@ -13,13 +13,12 @@
 - The scope of PR should be simple, unique and well-defined. PR should not contain unrelated changes
 - Approve PR only if you are sure about the scope
 - Be respectful and reply asap
-- Do not merge PR without approval
-- Avoid spending too much time on trivial changes
+- Avoid spending time on trivial changes
```
Comment (JoanFM) [inline .github/CODE_REVIEW_GUIDELINES.md:16]: Top much esa right; I think some minor style changes quickly add up

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: .github/CODE_REVIEW_GUIDELINES.md (hunk 9 above)
Comment (bhavsarpratik) [reply .github/CODE_REVIEW_GUIDELINES.md:16]: Didn't get you

Comment (JoanFM) [reply .github/CODE_REVIEW_GUIDELINES.md:16]: edited

**Comment by alexcg1** [comment]
Does anyone else have feedback? If not I'll merge this

**Review by JoanFM — APPROVED** [review]

**Review by alexcg1 — APPROVED** [review]
LGTM :+1:

**Comment by alexcg1** [comment]
@hanxiao any feedback? Looks like GitHub waiting on you since you're code owner

**Review by hanxiao — APPROVED** [review]

**Pull request merged**